tlatools/org.lamport.tlatools/src/tlc2/util/BufferedRandomAccessFile.tla

```diff
@@ -0,0 +1,384 @@
+\* Copyright (c) 2024, Oracle and/or its affiliates.
```

Comment:
IANAL: The copyright remains with Oracle, but the DCO makes sure that the TLA+ foundation may use the code in any way it likes within the bound of the open source license (MIT in our case).

Reply:
I think it would be worth emailing your contact with the Linux Foundation with this question.

Reply:
My company requires these notices on any code I write on company hardware and time. (Even for internal code!)

----

If you want more info than that...

I am not a lawyer, so I can't claim to fully understand the requirement, but here's what I know (or think I know anyway):
 - These notices do not change the project _license_ (MIT) in any way. Everyone is still free "to deal in the Software without restriction" and so forth. As you know, Oracle even signed a CLA attesting that it understands that fact.
 - These notices do not claim _exclusive_ copyright, and are compatible with overlapping copyrights from other organizations. Again, see the project license.
 - These notices primarily serve as glorified authorship information, and they might be best read by non-lawyers as meaning "this file contains code that was originally owned by Oracle and that Oracle is therefore legally allowed to release to the world under the terms of this project's license".

On a personal note, I find the notices aesthetically ugly. But, I do plan to follow the letter of the law so my company will let me keep contributing to TLA+ with company resources.

If the notices are unacceptable in new code, I can have a chat with my company's legal team and see if they'll accept something different. I know in the past they've been open to a [simple blanket copyright in the license file](https://github.com/wilcoxjay/mypyvy/blob/3a055dfad3d13fb35b6125ab6f43fa6ea4493b5f/LICENSE), but that could take a while to approve (and it might not ever be approved for a larger project like TLA+).

Reply:
I've been adding a similar Microsoft copyright notice in my contributions, and Microsoft also signed the CLA.  That's how it is done.